test/uk/gov/hmrc/gform/sharedmodel/SubmissionDataTest.scala

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright 2019 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.gform.sharedmodel
+
+import play.api.libs.json.Json
+import uk.gov.hmrc.gform.Spec
+
+class SubmissionDataTest extends Spec {
+
+  it should "serialized/deserialize in to/from json" in {
+    val submissionData = SubmissionData(
+      htmlForm,
+      Variables(
+        Map(
+          "usr.id"  -> "some value",
+          "usr.id2" -> "some value 2"
+        )))
+
+    Json.toJson(submissionData) should be(expectedJson)
+    expectedJson.as[SubmissionData] should be(submissionData)
+  }
+
+  val htmlForm = "<hmtl>something</html>"
+
+  val expectedJson =
+    Json.parse(
+      s"""{"pdfData":"$htmlForm","variables":{"value":{"usr.id":"some value","usr.id2":"some value 2"}}}""".stripMargin)
```

Comment:
Juts notice this. We are `stripMargin` from string which has no margin. It would be nice to pretty print that json.

Reply:
`stripMargin` removed, not sure pretty print because I'd need to pass a json and I like to see the string, brutal but clear.

Reply:
but this would work as well:
```
s"""|{
    |  "pdfData": "$htmlForm",
    |  "variables": {
    |    "value": {
    |      "usr.id": "some value",
    |      "usr.id2": "some value 2"
    |    }
    |  }
    |}""".stripMargin
```
and it is more readable.

Are you concerned about white spaces or what are you not seeing it this representation?

Reply:
Sure I missed understood I thought you mean call `pretty` and passed a json :-(

I do it now, I like it better too